src/views/UserInfo/index.vue

```diff
@@ -20,39 +19,85 @@ const message = useMessage()
 const [submitLoading, submitLoadingDispatcher] = useLoading()
 
 const formRef = ref<FormInst | null>(null)
+const uploadRef = ref<UploadInst | null>(null)
 const formData = ref<Partial<User>>({})
+const fileListLengthRef = ref({})
 
+const rules: FormRules = {
+  name: [
+    {
+      required: true,
+      message: '请输入用户名字',
+      trigger: ['blur', 'input']
+    }
+  ],
+  firstName: [
+    {
+      required: true,
+      message: '请输入名字',
+      trigger: ['blur', 'input']
+    }
+  ],
+  lastName: [
+    {
+      required: true,
+      message: '请输入姓',
+      trigger: ['blur', 'input']
+    }
+  ],
+  email: [
+    {
+      key: 'edit',
+      required: true,
+      trigger: ['blur', 'change'],
+      message: '请输入邮箱'
+    },
+    {
+      pattern: /^([a-zA-Z]|[0-9])(\w|-)+@[a-zA-Z0-9]+\.([a-zA-Z]{2,4})$/,
+      message: '请输入正确格式的邮箱',
+      trigger: ['input', 'blur']
+    }
+  ],
+  phoneNumber: [
+    {
+      pattern: /^[1][3456789]\d{9}$/,
+      message: '请输入正确格式的手机号',
+      trigger: ['input', 'blur']
+    }
+  ]
+}
 const computedUserInfo = computed(() => userStore.user)
 
 const handleValidateButtonClick = () => {
-  formRef.value?.validate((errors) => {
+  formRef.value?.validate(async (errors) => {
     if (errors) {
       message.error(errors[0][0].message!)
       return
     }
-
     if (submitLoading.value) {
       return
     }
 
     submitLoadingDispatcher.loading()
+    uploadRef.value?.submit()
+    await UploadApi.uploadFile(fileListLengthRef.value).then((res) => {
```

Comment:
`await` needs `try` and `catch` to process exceptions.

Reply:
ok